.github/workflows/bsi.yml

```diff
@@ -89,4 +104,44 @@ jobs:
           cache-key: bsi-${{ matrix.platform.host_os }}-${{ matrix.platform.compiler }}-${{ matrix.target }}
 
       - name: Build and Test Botan
-        run: python3 ./src/scripts/ci_build.py --with-bsi-policy --cc='${{ matrix.platform.compiler }}' --make-tool='${{ matrix.platform.make_tool }}' ${{ matrix.target }}
+        run: python3 ./src/scripts/ci_build.py --with-bsi-policy --cc='${{ matrix.platform.compiler }}' --make-tool='${{ matrix.platform.make_tool }}' --test-results-dir=junit_reports ${{ matrix.target }}
+
+      - name: Store JUnit Report
+        uses: actions/upload-artifact@v3
+        if: always()
+        with:
+          name: junit
+          path: junit_reports/*.xml
+          if-no-files-found: error
+
+  test_report:
+    name: "Generate Test Report"
+    needs: bsi_tests
+    if: always()
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Setup Build Agent
+        uses: ./.github/actions/setup-build-agent
+        with:
+          target: test-report
+
+      - name: Fetch JUnit Reports
+        uses: actions/download-artifact@v3
+        with:
+          name: junit
+          path: junit_reports
+
+      - name: Generate Test Report
+        run: |
+             python3 ./src/scripts/bsi_test_report.py --preamble doc/report_assets/testreport_preamble.rst --format=rst junit_reports test-report.rst
+             python3 ./src/scripts/bsi_test_report.py --preamble doc/report_assets/testreport_preamble.rst --format=pdf junit_reports test-report.pdf
+
+      - name: Store Test Report
+        uses: actions/upload-artifact@v3
+        if: always()
+        with:
+          name: Test Report
```

Comment:
Maybe add the short commit hash at the end. Same for the other artifacts. Makes it easier to remember to which state the downloaded artifacts belong.

Reply:
done

Reply:
Build failed due to a missing parenthesis in the script... 😞 trying again.

Reply:
![image](https://user-images.githubusercontent.com/1562139/203966409-a6375f6f-2b4e-46e0-9675-e44d3666fcff.png)

... same for the tarball.

Reply:
I was thinking about adding the short hash to the zip file names, but adding it to the internal names works for me too.